### PR TITLE
feat: unify all florence2 and sam2

### PR DESCRIPTION
**Pull request by Dayof**
# Description

Unifying the following florence2:
- florence2-ft
- florence2
- florence2-qa
- florence2-sam2

Adjusting the following tools with the new florence2 format:
- OCR
- TextToInstanceSegmentationTool
- TextToObjectDetection

Unifying all object-detection and segmentation response types:
- Tracking objects (sam2, florence2-sam2, TextToInstanceSegmentationTool)
```
[
    [
        {
            "id": int,
            "mask": rle,
            "bbox": [x_min, y_min, x_max, y_max],
            "label": str
        },
        {
            "id": int,
            "mask": rle,
            "bbox": [x_min, y_min, x_max, y_max],
            "label": str
        }
    ]
]
```

- Object detection with scores (owlv2, TextToObjectDetection)
```
[
    {
        "bboxes": [[x_min, y_min, x_max, y_max], [x_min, y_min, x_max, y_max]],
        "labels": [str, str],
        "scores": [float, float]
    }
]
```
- Object detection with scores (florence2, TextToObjectDetection)
```
[
    {
        "bboxes": [[x_min, y_min, x_max, y_max], [x_min, y_min, x_max, y_max]],
        "labels": [str, str]
    }
]
```
- QA (florence2-qa) (no batch)
```
{ "text": "" }
```
- QA (florence2) (batch)
```
[{ "text": "" }]
```
- OCR (florence2, OCR)
```
[
    {
        "quad_boxes": [[...], [...]],
        "labels": [str, str]
    }
]
```

This PR also refactors florence2-sam2 to use florence2 and sam2 models instead of having 2 different implementation for sam2.

# Extras

- Add notebooks to run florence2sam2 for video and image.
- Rename florencev2 to florence2.
- Add all bounding box postprocessing to florence2-sam2 + florence2 + owlv2.
- Add image batches to: florence2, florence2sam2, sam2, owlv2, OCR, TextToInstanceSegmentationRequest and TextToObjectDetectionRequest.
- Add video to: florence2, florence2sam2, sam2, owlv2, TextToInstanceSegmentationRequest and TextToObjectDetectionRequest.

# Tests

- florence2sam2 video (tracking car):
video: https://github.com/user-attachments/assets/0824ac08-7e90-4a6f-bb5b-8c475c780365
image

- florence2sam2 video (sharks):
video: https://github.com/user-attachments/assets/0f4c0a7a-829f-4903-8b9e-4fea51d41917
image

- florence2sam2 image:
image

# Performance

Shark video:

- Previous implementation for florence2sam2:
```
propagate in video: 100%|██████████| 21/21 [00:03<00:00,  5.58it/s]
propagate in video: 100%|██████████| 21/21 [00:04<00:00,  4.97it/s]
propagate in video: 100%|██████████| 21/21 [00:04<00:00,  4.88it/s]
propagate in video: 100%|██████████| 20/20 [00:04<00:00,  4.89it/s]
Time taken: 22.865721464157104 seconds
```

- Current refactor:
```
propagate in video: 100%|██████████| 21/21 [00:03<00:00,  5.61it/s]
propagate in video: 100%|██████████| 21/21 [00:04<00:00,  4.97it/s]
propagate in video: 100%|██████████| 21/21 [00:04<00:00,  4.87it/s]
propagate in video: 100%|██████████| 20/20 [00:04<00:00,  4.91it/s]
Time taken: 24.174272298812866 seconds
```

### CODE REVIEW

File: notebooks/run_florence2sam2_image.ipynb
Comment (CamiloInx) [inline notebooks/run_florence2sam2_image.ipynb:1]: @Dayof you might want to remove this file from the PR.

File: notebooks/run_florence2sam2_video.ipynb
Comment (CamiloInx) [inline notebooks/run_florence2sam2_video.ipynb:1]: Same with this file

Comment (Dayof) [reply notebooks/run_florence2sam2_video.ipynb:1]: why?

**Review by camiloaz — APPROVED** [review]

**Review by hrnn — APPROVED** [review]
overall it looks good to me, I think we can merge it once we fix the unit test.

File: docs/florence2-sam2.md [hunk 1]
```diff
@@ -25,19 +25,29 @@ cap.release()
...
+    ],
+    [
+        {
+            "id": 0
+            "mask": rle
+            "label": "ball"
+            "bbox": [x_min, y_min, x_max, y_max]
+        }
+    ]
+]
```
Comment (CamiloInx) [inline docs/florence2-sam2.md:50]: @Dayof aren't we moving towards [{bboxes:[], labels:[], scores:[]}]?

File: tests/models/florence2/data/models/caption_to_phrase_grounding_checkpoint.zip
Comment (CamiloInx) [inline tests/models/florence2/data/models/caption_to_phrase_grounding_checkpoint.zip:1]: @Dayof one question, why do we need to store these checkpoints? Is it needed for fine-tuning the models?

Comment (Dayof) [reply tests/models/florence2/data/models/caption_to_phrase_grounding_checkpoint.zip:1]: yes

**Pull request merged**